endpoints/cookie_sync.go

```diff
@@ -1,112 +1,166 @@
+	pbsAnalytics    analytics.PBSAnalyticsModule
+}
 
-		metrics.RecordCookieSync(pbsmetrics.Labels{})
-		userSyncCookie := usersync.ParsePBSCookieFromRequest(r, optOutCookie)
-		if !userSyncCookie.AllowSyncs() {
-			http.Error(w, "User has opted out", http.StatusUnauthorized)
-			co.Status = http.StatusUnauthorized
-			co.Errors = append(co.Errors, fmt.Errorf("user has opted out"))
-			return
-		}
+func (deps *cookieSyncDeps) Endpoint(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	//CookieSyncObject makes a log of requests and responses to  /cookie_sync endpoint
+	co := analytics.CookieSyncObject{
+		Status:       http.StatusOK,
+		Errors:       make([]error, 0),
+		BidderStatus: make([]*usersync.CookieSyncBidders, 0),
+	}
 
-		defer r.Body.Close()
-
-		csReq := &cookieSyncRequest{}
-		csReqRaw := map[string]json.RawMessage{}
-		err := json.NewDecoder(r.Body).Decode(&csReqRaw)
-		if err != nil {
-			if glog.V(2) {
-				glog.Infof("Failed to parse /cookie_sync request body: %v", err)
-			}
-			co.Status = http.StatusBadRequest
-			co.Errors = append(co.Errors, fmt.Errorf("JSON parse failed"))
-			http.Error(w, "JSON parse failed", http.StatusBadRequest)
-			return
-		}
-		biddersOmitted := true
-		if biddersRaw, ok := csReqRaw["bidders"]; ok {
-			biddersOmitted = false
-			err := json.Unmarshal(biddersRaw, &csReq.Bidders)
-			if err != nil {
-				if glog.V(2) {
-					glog.Infof("Failed to parse /cookie_sync request body (bidders list): %v", err)
-				}
-				co.Status = http.StatusBadRequest
-				co.Errors = append(co.Errors, fmt.Errorf("JSON parse failed (bidders"))
-				http.Error(w, "JSON parse failed (bidders)", http.StatusBadRequest)
-				return
-			}
-		}
+	defer deps.pbsAnalytics.LogCookieSyncObject(&co)
 
-		csResp := cookieSyncResponse{
-			BidderStatus: make([]*usersync.CookieSyncBidders, 0, len(csReq.Bidders)),
-		}
+	deps.metrics.RecordCookieSync(pbsmetrics.Labels{})
+	userSyncCookie := usersync.ParsePBSCookieFromRequest(r, deps.optOutCookie)
+	if !userSyncCookie.AllowSyncs() {
+		http.Error(w, "User has opted out", http.StatusUnauthorized)
+		co.Status = http.StatusUnauthorized
+		co.Errors = append(co.Errors, fmt.Errorf("user has opted out"))
+		return
+	}
 
-		if userSyncCookie.LiveSyncCount() == 0 {
-			csResp.Status = "no_cookie"
-		} else {
-			csResp.Status = "ok"
-		}
+	defer r.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		co.Status = http.StatusBadRequest
+		co.Errors = append(co.Errors, errors.New("Failed to read request body"))
+		http.Error(w, "Failed to read request body", http.StatusBadRequest)
+		return
+	}
+	biddersJSON, err := parseBidders(bodyBytes)
+	if err != nil {
+		co.Status = http.StatusBadRequest
+		co.Errors = append(co.Errors, errors.New("Failed to check request.bidders in request body. Was your JSON well-formed?"))
+		http.Error(w, "Failed to check request.bidders in request body. Was your JSON well-formed?", http.StatusBadRequest)
+	}
```

Comment:
Should there be a return here? If we are declaring an error condition in the response here, what use is further processing?